pkg/generate/generate.go

```diff
@@ -0,0 +1,160 @@
+package generate
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"moul.io/climan"
+	"moul.io/u"
+)
+
+func Cmd() *climan.Command {
+	var moduleRepo string
+	var outputDir string
+	var templateDir string
+	var protocBin string
+
+	return &climan.Command{
+		Name:       "generate",
+		ShortHelp:  "Generate code",
+		ShortUsage: "adapterkit generate [global flags] [flags] [args]",
+		FlagSetBuilder: func(fs *flag.FlagSet) {
+			fs.StringVar(&moduleRepo, "mod", "", "github repo where the module is located")
+			fs.StringVar(&outputDir, "out", ".", "output directory")
+			fs.StringVar(&templateDir, "tpl", "template", "template directory")
+			fs.StringVar(&protocBin, "protoc-bin", "protoc", "path to the 'protoc' binary")
+		},
+		Exec: func(_ context.Context, args []string) error {
+			if len(args) != 1 {
+				return flag.ErrHelp
+			}
+			if !u.CommandExists(protocBin) {
+				return fmt.Errorf("protoc binary not found: %s", protocBin) //nolint:goerr113
+			}
+
+			gotemplateOpts := fmt.Sprintf("--gotemplate_out=template_dir=%s,debug=true:%s", templateDir, outputDir)
+			protoPath := args[0]
+			cmd := exec.Command(protocBin, "-I.", gotemplateOpts, protoPath)
+			cmd.Stdout = os.Stdout
+			cmd.Stderr = os.Stderr
+			err := cmd.Run()
+			if err != nil {
+				log.Println("error: ", err)
+			}
+
+			currentModule, err := getModule(outputDir)
+			if err != nil {
+				return err
+			}
+
+			templateDirName, err := getTemplateDirName(templateDir)
+			if err != nil {
+				return err
+			}
+
+			err = fillImport(outputDir, moduleRepo, currentModule, templateDirName)
+			if err != nil {
+				return err
+			}
+
+			return nil
+		},
+	}
+}
+
+func getTemplateDirName(path string) (string, error) {
+	if len(path) == 0 {
+		return "", fmt.Errorf("empty path") //nolint:goerr113
+	}
+
+	// remove useless slash at the end of path
+	if path[len(path)-1] == '/' {
+		path = path[:len(path)-1]
+	}
+
+	tabPath := strings.Split(path, "/")
+	templateDirName := tabPath[len(tabPath)-1]
+
+	return templateDirName, nil
+}
+
+var errGetModule = errors.New("can't get module")
+
+func getModule(path string) (string, error) {
```

Comment:
fyi, this lib should help you to do it in a more official way; but let's stick to this implementation if you want.

golang.org/x/mod/modfile

Reply:
:+1: